File: crop_icon.py
import os
from PIL import Image

def main():
    filename = "恵比寿っち.jpg"
    if not os.path.exists(filename):
        print(f"File not found: {filename}")
        return
        
    img = Image.open(filename)
    print(f"Original size: {img.size}")
    
    # 3 patterns horizontally. We need to crop one. Usually the first one is at the left, or the center one.
    # Let's crop the first one (leftmost).
    width = img.width // 3
    height = img.height
    
    # Assuming the sprite is placed horizontally, we crop the middle frame because it might look best.
    # Or let's crop the left one. We can save all 3 and decide, but let's just create an icon from the first pattern.
    left = 0
    top = 0
    right = width
    bottom = height
    
    cropped = img.crop((left, top, right, bottom))
    
    # Also we want an icon to be square.
    # Let's see the size of one pattern.
    print(f"Cropped size: {cropped.size}")
    
    # Let's make it square 1024x1024 with some padding if necessary, or just scaling if it's already square.
    # We will pad it with white or extract its background color (top-left pixel).
    bg_color = cropped.getpixel((0, 0))
    
    # Create 1024x1024 background
    icon = Image.new("RGB", (1024, 1024), bg_color)
    
    # Resize cropped image to fit within 1024x1024 while preserving aspect ratio
    # Add a little padding to the sides so it's not sticking to the edge
    target_h = int(1024 * 0.8)
    scale = target_h / cropped.height
    target_w = int(cropped.width * scale)
    
    resized_cropped = cropped.resize((target_w, target_h), Image.LANCZOS)
    
    # Paste into center
    paste_x = (1024 - target_w) // 2
    paste_y = (1024 - target_h) // 2
    icon.paste(resized_cropped, (paste_x, paste_y))
    
    icon.save("icon.png")
    
    # Similarly, generate a splash screen (2732x2732)
    splash = Image.new("RGB", (2732, 2732), bg_color)
    target_h_splash = int(2732 * 0.4) # Splash image could be smaller relative to screen
    scale_splash = target_h_splash / cropped.height
    target_w_splash = int(cropped.width * scale_splash)
    
    resized_splash = cropped.resize((target_w_splash, target_h_splash), Image.LANCZOS)
    paste_x_s = (2732 - target_w_splash) // 2
    paste_y_s = (2732 - target_h_splash) // 2
    splash.paste(resized_splash, (paste_x_s, paste_y_s))
    
    splash.save("splash.png")
    print("Created icon.png and splash.png")

if __name__ == '__main__':
    main()

File: crop_icon.js
const { Jimp } = require('jimp');
const fs = require('fs');

async function main() {
    try {
        console.log('Loading image...');
        // Load the image
        const image = await Jimp.read('恵比寿っち.jpg');
        
        console.log(`Original Size: ${image.bitmap.width}x${image.bitmap.height}`);
        
        // The image has 3 frames horizontally. Let's calculate the width of one frame.
        const frameWidth = Math.floor(image.bitmap.width / 3);
        const frameHeight = image.bitmap.height;
        
        // Let's crop the center frame (or the left frame). The user said "any one of them".
        // The leftmost is often the default "idle" frame. Let's crop x=0.
        console.log(`Cropping to ${frameWidth}x${frameHeight}`);
        image.crop({ x: 0, y: 0, w: frameWidth, h: frameHeight });
        
        // Create an icon (1024x1024)
        console.log('Generating icon.png...');
        // Let's find background color from the top-left pixel
        const bgColorInt = image.getPixelColor(1, 1);
        
        // Create 1024x1024 blank canvas with bgColor
        const icon = new Jimp({ width: 1024, height: 1024, color: bgColorInt });
        
        // Resize cropped image to fit within the icon (leaving some padding so it looks good)
        // Let's make the height 80% of 1024
        const targetHeight = Math.floor(1024 * 0.8);
        const iconImage = image.clone();
        iconImage.resize({ h: targetHeight });
        
        // Center the image
        const xIcon = Math.floor((1024 - iconImage.bitmap.width) / 2);
        const yIcon = Math.floor((1024 - iconImage.bitmap.height) / 2);
        
        icon.composite(iconImage, xIcon, yIcon);
        await icon.write('icon.png');
        console.log('icon.png generated.');

        // Generate splash screen (2732x2732)
        console.log('Generating splash.png...');
        const splash = new Jimp({ width: 2732, height: 2732, color: bgColorInt });
        
        const splashTargetHeight = Math.floor(2732 * 0.4);
        const splashImage = image.clone();
        splashImage.resize({ h: splashTargetHeight });
        
        const xSplash = Math.floor((2732 - splashImage.bitmap.width) / 2);
        const ySplash = Math.floor((2732 - splashImage.bitmap.height) / 2);
        
        splash.composite(splashImage, xSplash, ySplash);
        await splash.write('splash.png');
        console.log('splash.png generated.');
        
    } catch (err) {
        console.error('Error during image processing:', err);
    }
}

main();
